.github/workflows/_test.yml

```diff
@@ -40,7 +40,9 @@ jobs:
         run: tox -e pytest
 
       - name: Upload coverage to Codecov
-        uses: codecov/codecov-action@v3
+        uses: codecov/codecov-action@v4
         with:
           name: ${{ inputs.python-version }}/${{ inputs.runs-on }}
           files: cov.xml
+        env:
+          CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}
```

Comment:
This is a secret, so you need to pass it into the workflow in the same way as you did for the other secrets in `ci.yaml`